Importing Libraries

In [1]:
#Import libraries
import pandas as pd
import numpy as np
import os

In [2]:
#import data - orders
df_ords = pd.read_csv(r'/Users/samuelcallender/Documents/ Instacart Basket Analysis/02 Data/Original Data/orders.csv', index_col = False)

In [3]:
#import data - products
df_prods = pd.read_csv(r'/Users/samuelcallender/Documents/ Instacart Basket Analysis/02 Data/Original Data/products.csv')


In [5]:
#import data - departments
df_dep = pd.read_csv(r'/Users/samuelcallender/Documents/ Instacart Basket Analysis/02 Data/Original Data/departments.csv')


In [6]:
#Find another identifier variable in the df_ords dataframe that doesn’t need to be included in your analysis as a numeric variable and change it to a suitable format.
df_ords['user_id'] = df_ords['user_id'].astype('str')

In [7]:
#Look for a variable in your df_ords dataframe with an unintuitive name and change its name without overwriting the data frame.
df_ords.rename(columns = {'order_dow' : 'order_day_of_week'}, inplace = True)

In [8]:
#Your client wants to know what the busiest hour is for placing orders. Find the frequency of the corresponding variable and share your findings.
df_ords['order_hour_of_day'].value_counts(dropna = False)

10    288418
11    284728
15    283639
14    283042
13    277999
12    272841
16    272553
9     257812
17    228795
18    182912
8     178201
19    140569
20    104292
7      91868
21     78109
22     61468
23     40043
6      30529
0      22758
1      12398
5       9569
2       7539
4       5527
3       5474
Name: order_hour_of_day, dtype: int64

The busiest hour is 10, with 288418 orders.

In [23]:
#Determine the meaning behind a value of 4 in the "department_id" column within the df_prods dataframe using a data dictionary.


In [25]:
df_prods = pd.read_csv(r'/Users/samuelcallender/Documents/ Instacart Basket Analysis/02 Data/Original Data/products.csv')

In [29]:
data_dict = df_prods.to_dict('index')

In [30]:
data_dict

{0: {'product_id': 1,
  'product_name': 'Chocolate Sandwich Cookies',
  'aisle_id': 61,
  'department_id': 19,
  'prices': 5.8},
 1: {'product_id': 2,
  'product_name': 'All-Seasons Salt',
  'aisle_id': 104,
  'department_id': 13,
  'prices': 9.3},
 2: {'product_id': 3,
  'product_name': 'Robust Golden Unsweetened Oolong Tea',
  'aisle_id': 94,
  'department_id': 7,
  'prices': 4.5},
 3: {'product_id': 4,
  'product_name': 'Smart Ones Classic Favorites Mini Rigatoni With Vodka Cream Sauce',
  'aisle_id': 38,
  'department_id': 1,
  'prices': 10.5},
 4: {'product_id': 5,
  'product_name': 'Green Chile Anytime Sauce',
  'aisle_id': 5,
  'department_id': 13,
  'prices': 4.3},
 5: {'product_id': 6,
  'product_name': 'Dry Nose Oil',
  'aisle_id': 11,
  'department_id': 11,
  'prices': 2.6},
 6: {'product_id': 7,
  'product_name': 'Pure Coconut Water With Orange',
  'aisle_id': 98,
  'department_id': 7,
  'prices': 4.4},
 7: {'product_id': 8,
  'product_name': "Cut Russet Potatoes Steam N' M

Value of 4: {'product_id': 5,
  'product_name': 'Green Chile Anytime Sauce',
  'aisle_id': 5,
  'department_id': 13,
  'prices': 4.3},

#The sales team in your client’s organization wants to know more about breakfast item sales. Create a subset containing only the required information.

In [32]:
df_breakfast = df_prods.loc[df_prods['department_id'] == 14]

In [33]:
df_breakfast

product_id                                       product_name  \
27             28                                  Wheat Chex Cereal   
33             34                                                NaN   
67             68                            Pancake Mix, Buttermilk   
89             90                                       Smorz Cereal   
210           211   Gluten Free Organic Cereal Coconut Maple Vanilla   
...           ...                                                ...   
49330       49326                            Cereal Variety Fun Pack   
49395       49391            Light and Fluffy Buttermilk Pancake Mix   
49547       49543                          Chocolate Cheerios Cereal   
49637       49633               Shake 'N Pour Buttermilk Pancake Mix   
49667       49663  Ultra Protein Power Crunch Peanut Butter N' Ho...   

       aisle_id  department_id  prices  
27          121             14    10.1  
33          121             14    12.2  
67          130             14    13.7  
89          121             14     3.9  
210         130             14     3.6  
...         ...            ...     ...  
49330       121             14     9.1  
49395       130             14     2.0  
49547       121             14    10.8  
49637       130             14    14.2  
49667        57             14    10.2  

[1116 rows x 5 columns]

#They’d also like to see details about products that customers might use to throw dinner parties. Your task is to find all observations from the entire dataframe that include items from the following departments: alcohol, deli, beverages, and meat/seafood. You’ll need to present this subset to your client.

In [34]:
df_dinner = df_prods.loc[df_prods['department_id'].isin([5,7,12,20])]

In [35]:
df_dinner

product_id                                    product_name  aisle_id  \
2               3            Robust Golden Unsweetened Oolong Tea        94   
6               7                  Pure Coconut Water With Orange        98   
9              10  Sparkling Orange Juice & Prickly Pear Beverage       115   
10             11                               Peach Mango Juice        31   
16             17                               Rendered Duck Fat        35   
...           ...                                             ...       ...   
49676       49672                          Cafe Mocha K-Cup Packs        26   
49679       49675             Cinnamon Dolce Keurig Brewed K Cups        26   
49680       49676                          Ultra Red Energy Drink        64   
49686       49682                              California Limeade        98   
49688       49684       Vodka, Triple Distilled, Twist of Vanilla       124   

       department_id  prices  
2                  7     4.5  
6                  7     4.4  
9                  7     8.4  
10                 7     2.8  
16                12    17.1  
...              ...     ...  
49676              7     6.5  
49679              7    14.0  
49680              7    14.5  
49686              7     4.3  
49688              5     5.3  

[7650 rows x 5 columns]

#It’s important that you keep track of total counts in your dataframes. How many rows does the last dataframe you created have?

7650 rows × 5 columns

#Someone from the data engineers team in Instacart thinks they’ve spotted something strange about the customer with a "user_id" of “1.” Extract all the information you can about this user.

In [36]:
df_ords_user1 = df_ords[df_ords['user_id']=='1']

In [37]:
df_ords_user1

order_id user_id eval_set  order_number  order_day_of_week  \
0    2539329       1    prior             1                  2   
1    2398795       1    prior             2                  3   
2     473747       1    prior             3                  3   
3    2254736       1    prior             4                  4   
4     431534       1    prior             5                  4   
5    3367565       1    prior             6                  2   
6     550135       1    prior             7                  1   
7    3108588       1    prior             8                  1   
8    2295261       1    prior             9                  1   
9    2550362       1    prior            10                  4   
10   1187899       1    train            11                  4   

    order_hour_of_day  days_since_prior_order  
0                   8                     NaN  
1                   7                    15.0  
2                  12                    21.0  
3                   7                    29.0  
4                  15                    28.0  
5                   7                    19.0  
6                   9                    20.0  
7                  14                    14.0  
8                  16                     0.0  
9                   8                    30.0  
10                  8                    14.0

#You also need to provide some details about this user’s behavior. What basic stats can you provide based on the information you have?

In [38]:
df_ords_user1.describe()

order_id  order_number  order_day_of_week  order_hour_of_day  \
count  1.100000e+01     11.000000          11.000000          11.000000   
mean   1.923450e+06      6.000000           2.636364          10.090909   
std    1.071950e+06      3.316625           1.286291           3.477198   
min    4.315340e+05      1.000000           1.000000           7.000000   
25%    8.690170e+05      3.500000           1.500000           7.500000   
50%    2.295261e+06      6.000000           3.000000           8.000000   
75%    2.544846e+06      8.500000           4.000000          13.000000   
max    3.367565e+06     11.000000           4.000000          16.000000   

       days_since_prior_order  
count               10.000000  
mean                19.000000  
std                  9.030811  
min                  0.000000  
25%                 14.250000  
50%                 19.500000  
75%                 26.250000  
max                 30.000000

#Export orders wrangled

In [41]:
path = r'/Users/samuelcallender/Documents/ Instacart Basket Analysis/02 Data/Prepared Data'

In [44]:
path

'/Users/samuelcallender/Documents/ Instacart Basket Analysis/02 Data/Prepared Data'

In [50]:
df_ords.to_csv(r'/Users/samuelcallender/Documents/ Instacart Basket Analysis/02 Data/Prepared Data/orders_wrangled.csv')